src/main/java/sparksoniq/spark/iterator/flowr/CountClauseSparkIterator.java

```diff
@@ -109,4 +111,30 @@ private void setNextLocalTupleResult() {
                 .mapValues(index -> index + 1)
                 .map(new CountClauseClosure(variableName, getMetadata()));
     }
+
+    public Set<String> getVariableDependencies()
+    {
+        Set<String> result = new HashSet<String>();
+        result.addAll(_child.getVariableDependencies());
```

Comment:
I just wanted to point out that the following pattern is not applied to the CountClause. Why are the child's bound variables not removed here?
        if(_child != null)
        {
            result.removeAll(_child.getBoundVariables());
            result.addAll(_child.getVariableDependencies());
        }

Additionally, could you elaborate on the algorithm here a bit?